patches/patch-sharpziplib-ca-61725.diff

```diff
@@ -1,62 +0,0 @@
-# Copyright (c) Cloud Software Group, Inc.
-# 
-# Redistribution and use in source and binary forms,
-# with or without modification, are permitted provided
-# that the following conditions are met:
-# 
-# *   Redistributions of source code must retain the above
-#     copyright notice, this list of conditions and the
-#     following disclaimer.
-# *   Redistributions in binary form must reproduce the above
-#     copyright notice, this list of conditions and the
-#     following disclaimer in the documentation and/or other
-#     materials provided with the distribution.
-# 
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
-# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
-# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
-# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
-# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
-# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
-# SUCH DAMAGE.
-
---- a/src/Tar/TarOutputStream.cs	2007-09-07 12:19:20.000000000 +0100
-+++ b/src/Tar/TarOutputStream.cs	2011-07-08 13:28:44.804190600 +0100
-@@ -266,7 +266,9 @@
- 				throw new ArgumentNullException("entry");
- 			}
- 
--			if (entry.TarHeader.Name.Length >= TarHeader.NAMELEN) {
-+			int byteCount = Encoding.UTF8.GetByteCount(entry.TarHeader.Name);
-+
-+			if (byteCount >= TarHeader.NAMELEN) {
- 				TarHeader longHeader = new TarHeader();
- 				longHeader.TypeFlag = TarHeader.LF_GNU_LONGNAME;
- 				longHeader.Name = longHeader.Name + "././@LongLink";
-@@ -276,16 +278,16 @@
- 				longHeader.UserName = "";
- 				longHeader.LinkName = "";
- 
--				longHeader.Size = entry.TarHeader.Name.Length;
-+				longHeader.Size = byteCount;
- 
- 				longHeader.WriteHeader(this.blockBuffer);
- 				this.buffer.WriteBlock(this.blockBuffer);  // Add special long filename header block
- 
- 				int nameCharIndex = 0;
- 
--				while (nameCharIndex < entry.TarHeader.Name.Length) {
-+				while (nameCharIndex < byteCount) {
- 					Array.Clear(blockBuffer, 0, blockBuffer.Length);
--					TarHeader.GetAsciiBytes(entry.TarHeader.Name, nameCharIndex, this.blockBuffer, 0, TarBuffer.BlockSize);
-+					TarHeader.GetNameBytes(entry.TarHeader.Name, nameCharIndex, this.blockBuffer, 0, TarBuffer.BlockSize);

```

Comment:
As discussed separately, there is a risk here:
- our patch uses longHeader.Size = byteCount;, while their code adds 1
- our patch uses TarHeader.GetNameBytes, while their code uses TarHeader.GetAsciiBytes passing in the additional parameter